### PR TITLE
[Test][Worker] Add CPU UT coverage for Kimi K2.5 patch

### PR DESCRIPTION
**Pull request by Dawn952**
### What this PR does / why we need it?

Adds CPU-only unit-test coverage for `patch_kimi_k25.py`.

The tests exercise the live upstream vLLM contracts used by the patch instead of mocking those interfaces:

- instantiate `Learnable2DInterpPosEmbDivided_fixed` with its current upstream constructor and call the patched `forward` with both list and tensor grid inputs;
- verify the patch is installed on the upstream class;
- execute the A5 branch on CPU and verify `MoonViT3dPretrainedModel.to()` ignores both positional and keyword `dtype` arguments while preserving parameter dtype.

This makes main2main upgrade PRs fail in CPU UT when the related upstream constructor, decorator, class, or call contract becomes incompatible, rather than discovering the breakage later in nightly model tests.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- `uvx ruff check tests/ut/patch/worker/test_patch_kimi_k25.py`
- `uvx ruff format --check tests/ut/patch/worker/test_patch_kimi_k25.py`
- `python -m py_compile tests/ut/patch/worker/test_patch_kimi_k25.py`
- `git diff --check`
- verified `.github/workflows/scripts/select_tests.py` routes the new test to the CPU runner
- repository CPU selected-tests CI (pending maintainer-applied `ready` label)

- vLLM version: v0.26.0
- vLLM main: https://github.com/vllm-project/vllm/commit/d02df748bf9efd99022f1a062597dc3cb3808485

### CODE REVIEW

File: tests/ut/patch/worker/test_patch_kimi_k25.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        vision_tower = MoonViT3dPretrainedModel.__new__(MoonViT3dPretrainedModel)
+        nn.Module.__init__(vision_tower)
+        vision_tower.register_parameter(
+            "quantized_weight",
+            nn.Parameter(torch.ones(1, dtype=torch.float32)),
+        )
+
+        assert vision_tower.to(torch.float16) is vision_tower
+        assert vision_tower.to(dtype=torch.bfloat16) is vision_tower
+        assert vision_tower.quantized_weight.dtype == torch.float32
```
[the remark this answers is not in the document]
Comment (Dawn952) [reply tests/ut/patch/worker/test_patch_kimi_k25.py]: Accepted, thanks. Moved `runpy.run_path` inside the existing `try/finally`, so both upstream class methods are restored even if executing the patch module raises. Addressed in 175c2ba86.

**Review by yiz-liu — APPROVED** [review]
